src/main/java/vivatech/util/TierHelper.java

```diff
@@ -0,0 +1,45 @@
+package vivatech.util;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import net.minecraft.block.Block.Settings;
+import net.minecraft.item.BlockItem;
+import net.minecraft.item.Item;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import vivatech.Vivatech;
+import vivatech.block.AbstractTieredMachineBlock;
+
+public class TierHelper {
+
+	public static Identifier getTieredID(Identifier id, MachineTiers tier) {
+		return new Identifier(id.getNamespace(), tier.getAffix() + "_" + id.getPath());
+	}
+	
+	public static AbstractTieredMachineBlock[] getBlockArray(Map<MachineTiers, ? extends AbstractTieredMachineBlock> map) {
+		return map.values().toArray(new AbstractTieredMachineBlock[MachineTiers.values().length]);
+	}
+
+	public static void registerTieredBlockMap(Map<MachineTiers, ? extends AbstractTieredMachineBlock> map) {
```

Comment:
Why not `registerTieredBlocks(AbstractTieredMachineBlock... blocks)` since each block knows what tier it is?